sdk/src/wallet/core/mod.rs

```diff
@@ -313,6 +313,16 @@ impl WalletData {
             .filter(|output_data| output_data.output.is_account())
     }
 
+    // Returns the first possible Account id, which can be an implicit account.
+    // If none are found, returns AccountNotFound.
+    pub fn first_account_id(&self) -> Result<AccountId> {
```

Comment:
Actually, I think this should just return `Option<AccountId>` and the usage should be like

```
let account_id = account_id.or_else(|| wallet_data.first_account_id()).ok_or(Error::AccountNotFound)?;
```

Reply:
I agree